test/Api.Test/Vault/AuthorizationHandlers/GroupAuthorizationHandlerTests.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.Security.Claims;
+using Bit.Api.Vault.AuthorizationHandlers.Groups;
+using Bit.Core;
+using Bit.Core.Context;
+using Bit.Core.Enums;
+using Bit.Core.Models.Data;
+using Bit.Core.Test.AutoFixture;
+using Bit.Test.Common.AutoFixture;
+using Bit.Test.Common.AutoFixture.Attributes;
+using Microsoft.AspNetCore.Authorization;
+using NSubstitute;
+using Xunit;
+
+namespace Bit.Api.Test.Vault.AuthorizationHandlers;
+
+[SutProviderCustomize]
+[FeatureServiceCustomize(FeatureFlagKeys.FlexibleCollections)]
+public class GroupAuthorizationHandlerTests
+{
+    [Theory]
+    [BitAutoData(OrganizationUserType.Admin, false, false, false, false, false, true)]
+    [BitAutoData(OrganizationUserType.Owner, false, false, false, false, false, true)]
+    [BitAutoData(OrganizationUserType.User, false, false, false, false, false, false)]
+    [BitAutoData(OrganizationUserType.Custom, true, false, false, false, false, true)]
+    [BitAutoData(OrganizationUserType.Custom, false, true, false, false, false, true)]
+    [BitAutoData(OrganizationUserType.Custom, false, false, true, false, false, true)]
+    [BitAutoData(OrganizationUserType.Custom, false, false, false, true, false, true)]
+    [BitAutoData(OrganizationUserType.Custom, false, false, false, false, true, true)]
+    [BitAutoData(OrganizationUserType.Custom, false, false, false, false, false, false)]
+    public async Task CanReadAllAccessAsync_ReturnsExpectedResult(
+        OrganizationUserType userType, bool editAnyCollection, bool deleteAnyCollection,
+        bool manageGroups, bool manageUsers, bool accessImportExport, bool expectedSuccess,
+        Guid userId, SutProvider<GroupAuthorizationHandler> sutProvider,
+        CurrentContextOrganization organization)
```

Comment:
This is really hard to read and maintain. I'd rather some test code repetition than a big matrix of bools like this.

I also think that an `expectedSuccess` bool is a bit of a smell. Better to have separate test methods for separate test cases.

To be clear, `BitAutoData` is still really good and useful, I just think this leans on it too heavily.

Suggestion for how you might improve this (you might think of a better way):
* `CanReadAllAccessAsync_WhenAdminOrOwner_Success`
* `CanReadAllAccessAsync_WhenUser_Fail`
* `CanReadAllAccessAsync_WhenCustomUserWithRequiredPermissions_Success`

Reply:
I have split this test, it should be easier to read/maintain

Reply:
I think these tests look really good now!